source/slang/options.cpp

```diff
@@ -566,6 +577,25 @@ struct OptionsParser
                         compileRequest,
                         passThrough);
                 }
+                else if (argStr == "-dxc-path")
+                {
+                    String name;
+                    SLANG_RETURN_ON_FAIL(tryReadCommandLineArgument(sink, arg, &argCursor, argEnd, name));
+                    addSharedLibraryPath(SharedLibraryType::Dxc, name);
+                    addSharedLibraryPath(SharedLibraryType::Dxil, name);
```

Comment:
Just to make sure I'm understanding: the user is specifying the directory path, and not the path to the DLL itself?

I might actually argue that `-dxc-path` seems like it would be specifying the path to `dxcompiler.dll` itself (or maybe even `dxc.exe`, but lets hope users don't get *that* confused). I might advocate for changing this to `-dxc-directory` or `-dxc-directory-path` if it is going to have the current semantics.

Alternatively a very "do what I mean" option would be to check if the supplied string names a file or a directory, and react accordingly. In the case where a file path is specified for dxc, we could then split off the directory name and use that for looking up the DXIL dll.

Reply:
One more note here is that maybe we want to support a `-dxil-path` option, to specify the location for that DLL separately, and then have logic *after* the command line options are all parsed and implicitly set a path for looking up `dxil.dll` based on the dxc path, if one wasn't set explicitly.

Reply:
On dxil-path, I mean I can do, I did it this way for simplicity reasons, and it seemed somewhat unlikely that someone would want them in different places. 

The user is specifying the directory path and not the dll. Doing so enables the behavior above, and the actual specification of the name is typically somewhat redundant. 

I understand the original point tho. -dxc-directory-path seems kind of long, so perhaps -dxc-directory is enough to convey the meaning better.